lib/inspec/reporters/cli.rb

```diff
@@ -166,9 +166,9 @@ def format_with_color(color_name, text)
     end
 
     def all_unique_controls
-      return @unique_controls unless @unique_controls.nil?
+      @unique_controls ||= Set.new
+      return @unique_controls unless @unique_controls.empty?
 
-      @unique_controls = Set.new
       run_data[:profiles].each do |profile|
         profile[:controls].map { |control| @unique_controls.add(control) }
       end
```

Comment:
This one I'm gonna turn into an opportunity to evolve this. Getting rid of the warning is absolutely desired, but this method is a bit of a mess (not on you at all). But! Now we have the opportunity to clean this up!

1) original
```ruby
    def all_unique_controls
      return @unique_controls unless @unique_controls.nil?

      @unique_controls = Set.new
      run_data[:profiles].each do |profile|
        profile[:controls].map { |control| @unique_controls.add(control) }
      end

      @unique_controls
    end
```
2) don't access uninitialized ivar (yay!) This is definitely one way to do it. Another is to simply set it in `initialize`, but the point of this method is to be a "lazy accessor" so it doesn't compute the set until it is used.
```ruby
    def all_unique_controls
      @unique_controls ||= Set.new
      return @unique_controls unless @unique_controls.empty?

      run_data[:profiles].each do |profile|
        profile[:controls].map { |control| @unique_controls.add(control) }
      end

      @unique_controls
    end
```
3) New let's focus on the content. That map stands out as very odd since it isn't going anywhere. So don't map if you're not mapping (tho, we are, as we will see below)
```ruby
    def all_unique_controls
      @unique_controls ||= Set.new
      return @unique_controls unless @unique_controls.empty?

      run_data[:profiles].each do |profile|
        profile[:controls].each do |control|
          @unique_controls.add(control)
        end
      end

      @unique_controls
    end
```
4) Use idiomatic caching method structure: `@x ||= begin ... end`
```ruby
    def all_unique_controls
      @unique_controls ||= begin
                             uc = Set.new

                             run_data[:profiles].each do |profile|
                               profile[:controls].each do |control|
                                 uc.add(control)
                               end
                             end

                             uc
                           end
    end
```
5) Finally, now that we can see what this is: a lazy accessor that is trying to return a unique collection of profile controls, let's strip it back to the essentials -- all this is really doing is getting the profile's controls (read: a map) and returning a unique collection (read: .uniq) of them, so just do that:
```ruby
    def all_unique_controls
      @unique_controls ||= begin
                             run_data[:profiles].map { |profile|
                               profile[:controls]
                             }.uniq
                           end
    end
```

Reply:
Perfect feedback. Thank you!